rest_framework_json_api/schemas/openapi.py

```diff
@@ -0,0 +1,878 @@
+            if hasattr(view, 'action'):
+                current_action = view.action
+                view.action = action
+            operation = view.schema.get_operation(path, method, action)
+            components = view.schema.get_components(path, method)
+            for k in components.keys():
+                if k not in components_schemas:
+                    continue
+                if components_schemas[k] == components[k]:
+                    continue
+                warnings.warn(
+                    'Schema component "{}" has been overriden with a different value.'.format(k))
+
+            components_schemas.update(components)
+
+            if hasattr(view.schema, 'get_security_schemes'):  # pragma: no cover
+                security_schemes = view.schema.get_security_schemes(path, method)
+            else:
+                security_schemes = {}
+            for k in security_schemes.keys():  # pragma: no cover
+                if k not in security_schemes_schemas:
+                    continue
+                if security_schemes_schemas[k] == security_schemes[k]:
+                    continue
+                warnings.warn('Securit scheme component "{}" has been overriden with a different '
+                              'value.'.format(k))
+            security_schemes_schemas.update(security_schemes)  # pragma: no cover
+
+            if hasattr(view, 'action'):
+                view.action = current_action
+            # Normalise path for any provided mount url.
+            if path.startswith('/'):
+                path = path[1:]
+            path = urljoin(self.url or '/', path)
+
+            paths.setdefault(path, {})
+            paths[path][method.lower()] = operation
+
+        self.check_duplicate_operation_id(paths)
+
+        # Compile final schema, overriding stuff from super class.
+        schema['paths'] = paths
+        schema['components'] = self.jsonapi_components
+        schema['components']['schemas'].update(components_schemas)
+        if len(security_schemes_schemas) > 0:  # pragma: no cover
+            schema['components']['securitySchemes'] = security_schemes_schemas
+
+        return schema
+
+    def _expand_relationships(self, path, method, view):
+        """
+        Expand path containing .../{id}/relationships/{related_field} into list of related fields.
+        :return:list[tuple(path, method, view, action)]
+        """
+        queryset = view.get_queryset()
+        if not queryset.model:
+            return [(path, method, view, getattr(view, 'action', '')), ]
+        result = []
+        # TODO: what about serializer-only (non-model) fields?
+        #       Shouldn't this be iterating over serializer fields rather than model fields?
+        #       Look at parent view's serializer to get the list of fields.
+        #       OR maybe like _expand_related?
+        m = queryset.model
+        for field in [f for f in dir(m) if not f.startswith('_')]:
```

Comment:
@sliverc I have to research this one some more. The declaration of this RelationShipView is dynamic, for example:
```
    url(r'^entries/(?P<pk>[^/.]+)/relationships/(?P<related_field>\w+)$',
        EntryRelationshipView.as_view(),
        name='entry-relationships'),
```

But those relationship URLs are produced by the EntryView so I have to find the way back to the EntryView given all I have from the EntryRelationshipView is the Entry model:
```
class EntryRelationshipView(RelationshipView):
    queryset = Entry.objects.all()
```

And this certainly is incorrect right now because the Entry schema generated looks like this (with 10 relationships):
```yaml
    Entry:
      type: object
      required:
      - type
      - id
      additionalProperties: false
      properties:
        type:
          $ref: '#/components/schemas/type'
        id:
          $ref: '#/components/schemas/id'
        links:
          type: object
          properties:
            self:
              $ref: '#/components/schemas/link'
        attributes:
          type: object
          properties:
            headline:
              type: string
              maxLength: 255
            body_text:
              type: string
              nullable: true
            pub_date:
              type: string
              format: date
              nullable: true
            mod_date:
              type: string
              format: date
              nullable: true
            body_format:
              type: string
              readOnly: true
          required:
          - headline
        relationships:
          type: object
          properties:
            blog:
              $ref: '#/components/schemas/reltoone'
            blog_hyperlinked:
              $ref: '#/components/schemas/reltoone'
            authors:
              $ref: '#/components/schemas/reltomany'
            comments:
              $ref: '#/components/schemas/reltomany'
            comments_hyperlinked:
              $ref: '#/components/schemas/reltomany'
            featured:
              $ref: '#/components/schemas/reltoone'
            suggested:
              $ref: '#/components/schemas/reltoone'
            suggested_hyperlinked:
              $ref: '#/components/schemas/reltoone'
            tags:
              $ref: '#/components/schemas/reltomany'
            featured_hyperlinked:
              $ref: '#/components/schemas/reltoone'
```

but the relationships paths only have those (3) from the Model:
```
/entries/{id}/relationships/authors
/entries/{id}/relationships/comments
/entries/{id}/relationships/tags
```

BTW, the command to generate the schema is:
`django-admin generateschema --generator_class rest_framework_json_api.schemas.openapi.SchemaGenerator --settings=example.settings`

Maybe the right approach to this is to make use of the EntryViewSet to generate the `relationships` paths and ignore the EntryRelationshipView.

I'll dig some more.

Reply:
Maybe I've overthought this and can just put a templated URL in the schema:
```yaml
paths:
  /entries/{id}/relationships/{related_field}:
    get:
      operationId: List/entries/{id}/relationships/{related_field}
      description: ''
      parameters:
      - name: id
        in: path
        required: true
        description: A unique integer value identifying this entry.
        schema:
          type: string
      - name: related_field
        in: path
        required: true
        description: the name of the related field
        schema:
          type: string
```

There's no real reason to have to expand out all the possible related field names here.... And, this URL will likely only be used via HATEOAS link from the `/entries/{id}` response.

Reply:
... and a future enhancement would be to change the `related_field` to include an enumeration of allowed values (probably via $ref to components/parameters/Entry_related_fields):

```
      - name: related_field
        in: path
        required: true
        description: the name of the related field
        schema:
          type: string
          enum:
            - blog
            - blog_hyperlinked
            - authors
            - comments
            - comments_hyperlinked
            - featured
            - suggested
            - suggested_hyperlinked
            - tags
            - featured_hyperlinked
```

Reply:
This makes a nicer swagger-ui:

<img width="604" alt="Screen Shot 2020-10-12 at 11 29 08 AM" src="https://user-images.githubusercontent.com/4356471/95764532-96c77a00-0c7e-11eb-89ad-80a289f793cc.png">

Reply:
On further review I think I need to revert 5855b65 and revisit it since relationships can be either singular or plural and that depends on the individual relationship definitions.

Reply:
I like how handling of `RelationshipView` has been simplified now.